src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -10,21 +10,47 @@ const CanonicalChartbeatBeacon = ({
   chartbeatUID,
   useCanonical,
   chartbeatSource,
-}) => (
-  <Helmet>
-    <script async type="text/javascript">
-      {`
+  hasCookie,
+  title,
+  referrer,
+  hasReferrer,
+}) => {
+  const optionalConfigValues = {
+    ...(hasReferrer && { virtualReferrer: referrer }),
+    ...(hasCookie && { idSync: { bbc_hid: cookie } }),
```

Comment:
Just realised this duplicates https://github.com/bbc/simorgh/pull/2837/files#diff-a6cb5242b50b07228921447ee534136aR11-R12

I think it would make more sense to move any configuration that is common to both AMP and canonical into https://github.com/bbc/simorgh/blob/latest/src/app/containers/ChartbeatAnalytics/index.jsx

Reply:
yeah, @jamesdonoh we were doing that before but one of the things we needed to change was sending null values, so if we move those conditional logic down one prop level they'll be defined as null where their being used

Reply:
@jamesdonoh  on the ticket description [here](https://github.com/bbc/simorgh/issues/2811), it says `If a user is not signed in, we are sending 'null'. We should not send this parameter at all.`